cpp/src/arrow/compute/kernels/scalar_string.cc

```diff
@@ -93,19 +93,23 @@ struct BinaryLength {
   }
 };
 
+template <typename OffsetValue>
+OffsetValue CountCodepoints(const uint8_t* str, size_t strlen) {
```

Comment:
There's no need to template this on `OffsetValue`, just return `int64_t`.
Also, since this is a generally useful function, perhaps put it in `arrow/util/utf8.h`?

Reply:
Moved & added a small test case.